src/SFA.DAS.EAS.Employer_Financial.Database/StoredProcedures/GetEffectivePeriod12Declaration.sql

```diff
@@ -0,0 +1,32 @@
+﻿CREATE PROCEDURE [employer_financial].[GetEffectivePeriod12Declaration]
+	@EmpRef varchar(50),
+	@PayrollYear varchar(10),
+	@YearEndAdjustmentCutOff DateTime
+AS
+	SELECT TOP 1
+		Id, 
+		AccountId, 
+		EmpRef, 
+		LevyDueYTD, 
+		LevyAllowanceForYear, 
+		SubmissionDate, 
+		SubmissionId, 
+		PayrollYear, 
+		PayrollMonth, 
+		CreatedDate, 
+		EndOfYearAdjustment,
+		EndOfYearAdjustmentAmount,
+		LevyAllowanceForYear,
+		DateCeased,
+		InactiveFrom,
+		InactiveTo,
+		HmrcSubmissionId,
+		NoPaymentForPeriod
+	FROM 
+		[employer_financial].[GetLevyDeclaration]
+	WHERE	EmpRef = @EmpRef
+			AND	PayrollYear = @PayrollYear
+			AND (LastSubmission = 1 OR (EndOfYearAdjustment = 1 AND SubmissionDate < @yearEndAdjustmentCutOff))
```

Comment:
I'm being lazy...do we need to make sure the table has an appropriate index?

Reply:
The query is narrower than the sproc used previously by the repo but I've checked and there is a covering index on empref and payroll year. Given that an empref would be expected to typically make 12 declarations per year that should be fine.